src/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/MyServices/Internal/ContextValue.vb

```diff
@@ -0,0 +1,115 @@
+' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Option Explicit On
+Option Strict Off
+
+Imports System.Collections
+Imports System.Collections.Generic
+Imports System.Threading
+
+Namespace Microsoft.VisualBasic.MyServices.Internal
+
+    '''**************************************************************************
+    ''' ;SkuSafeHttpContext
+    ''' <summary>
+    ''' Returns the current HTTPContext or nothing if we are not running in a 
+    ''' web context.
+    ''' </summary>
+    ''' <remarks>
+    ''' With the FX dividing into Client and Full skus, we may not always have
+    ''' access to the System.Web types.  So we have to test for the presence
+    ''' of System.Web.Httpcontext before trying to access it.
+    ''' </remarks>
+    <Global.System.ComponentModel.EditorBrowsableAttribute(Global.System.ComponentModel.EditorBrowsableState.Never)>
+    Friend Class SkuSafeHttpContext
+        Public Shared ReadOnly Property Current() As Object
+            Get
+                'Return the equivalent of System.Web.HttpContext.Current
+                If m_HttpContextCurrent IsNot Nothing Then
+                    Return m_HttpContextCurrent.GetValue(Nothing, Nothing)
+                Else
+                    Return Nothing
+                End If
+            End Get
+        End Property
+
+        '''**************************************************************************
+        ''' ;InitContext
+        ''' <summary>
+        ''' Initialize the field that holds the type that allows us to access
+        ''' System.Web.HttpContext
+        ''' </summary>
+        Private Shared Function InitContext() As System.Reflection.PropertyInfo
+            Dim HttpContextType As System.Type
+            HttpContextType = System.Type.GetType("System.Web.HttpContext,System.Web,Version=0.0.0.0,Culture=neutral,PublicKeyToken=B03F5F7F11D50A3A")
```

Comment:
Will we ever have this type on .NET Core?

Reply:
No

Reply:
Thanks. In that case, I'll remove this type.

---
In reply to: [310743374](https://github.com/dotnet/corefx/pull/40036#discussion_r310743374) [](ancestors = 310743374)